### PR TITLE
General Code Refactoring

### PR DESCRIPTION
**Pull request by BlairArchibald**
Based on an intellij code analysis pass.

I've based this on CareHomes (assuming that will probably go in first)

### CODE REVIEW

File: src/main/java/uk/co/ramp/covid/simulation/covid/Covid.java [hunk 1]
```diff
@@ -103,26 +102,30 @@ private void setSymptomatic() {
...
 
     public CStatus stepInfection(Time t) {
-    	CStatus status = null;
-    	if(symptomaticCase) status = this.stepInfectionSymptomatic(t);
-    	else if(!symptomaticCase) status = this.stepInfectionAsymptomatic();
-    	return status;
+    	if(symptomaticCase) {
+    	    return this.stepInfectionSymptomatic(t);
+        }
+    	return this.stepInfectionAsymptomatic();
```
Comment (johnnonweiler) [inline src/main/java/uk/co/ramp/covid/simulation/covid/Covid.java:128]: From a style perspective, I'd prefer an explicit _else_ here.  (However, this is fine, and I'm not asking you to change it.)

File: src/main/java/uk/co/ramp/covid/simulation/parameters/PopulationParameters.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ public boolean isValid() {
         boolean valid = true;
         // We don't do this in a single statement to ensure that all the "uninitalised" parameter warnings are printed
         // in one go instead of being short circuited
-        valid = valid && checker.isValid(populationDistribution);
+        valid = checker.isValid(populationDistribution);
```
Comment (johnnonweiler) [inline src/main/java/uk/co/ramp/covid/simulation/parameters/PopulationParameters.java:44]: If you're changing this, then it doesn't make sense anymore to have line 41 setting `valid = true`.  (I'm a little surprised that IntelliJ didn't complain about this!)
```suggestion
        boolean valid = checker.isValid(populationDistribution);
```

File: src/test/java/uk/co/ramp/covid/simulation/integrationTests/MovementTest.java [hunk 3]
```diff
@@ -445,17 +446,17 @@ public void positiveTestsStayInQuarantine() {
 
         per.infect();
         per.getcVirus().forceSymptomatic(true);
-        // Usually there's a delay before symptonms but we just force it here
-        double time = per.getcVirus().getSymptomDelay() + 1;
-        for (int j = 0; j < time; j++) {
+
+        //  Usually there's a delay before symptonms but we just force it here
+        while (!per.getcVirus().isSymptomatic()) {
```
Comment (johnnonweiler) [inline src/test/java/uk/co/ramp/covid/simulation/integrationTests/MovementTest.java:451]: This looks dangerous.  If something goes wrong, this test could get stuck in an infinite loop.

File: src/main/java/uk/co/ramp/covid/simulation/covid/Covid.java (hunk 1 above)
Comment (BlairArchibald) [reply src/main/java/uk/co/ramp/covid/simulation/covid/Covid.java:128]: Thanks! I'm also a bit surprised - maybe I need to hit some form of refresh button I don't know about!

File: src/test/java/uk/co/ramp/covid/simulation/integrationTests/MovementTest.java (hunk 3 above)
Comment (BlairArchibald) [reply src/test/java/uk/co/ramp/covid/simulation/integrationTests/MovementTest.java:451]: I agree - I'm planning on adding some features (probably in another PR) that gets the number of steps we need to advance from the COVID object itself.

There's a couple of these types of loops that have bitten in the past.

**Comment by BlairArchibald** [comment]
I think this is almost ready to go in - but some of the conditionals etc that were in the merge looked a bit strange (mainly !inf.isSymptomatic was inf.isSympotmatic) so it would be nice to have a second set of eyes before we merge this in. The refactorings themselves are the same as before with a few typo's fixed etc

File: src/test/java/uk/co/ramp/covid/simulation/place/CareHomeTest.java [hunk 4]
```diff
@@ -110,17 +110,22 @@ public void sickResidentsAreQuarantined() {
         Time t = new Time(0);
         while (!inf.getcVirus().isSymptomatic()) {
             inf.getcVirus().stepInfection(t);
+            inf.cStatus();
             t = t.advance();
         }
-        
+
         // Not quarantined instantly
+        // TODO You might have symptoms but have a trans adjustment of 0 (if you get symptom while latent).
```
Comment (johnnonweiler) [inline src/test/java/uk/co/ramp/covid/simulation/place/CareHomeTest.java:118]: Should we add a GitHub issue to fix this new TODO?

File: src/main/java/uk/co/ramp/covid/simulation/place/CareHome.java [hunk 5]
```diff
@@ -9,13 +9,12 @@
...
 
-    private RoundRobinAllocator<Shifts> shifts;
-    private boolean isQuarantined;
+    private final RoundRobinAllocator<Shifts> shifts;
 
     public CareHome(Size s) {
         super(s);
 
-        transAdjustment = PopulationParameters.get().buildingProperties.careHomeTransmissionConstant;
+        transConstant = PopulationParameters.get().buildingProperties.careHomeTransmissionConstant;
```
Comment (johnnonweiler) [inline src/main/java/uk/co/ramp/covid/simulation/place/CareHome.java:17]: This is not refactoring, and it's not obvious why it's here.

File: src/main/java/uk/co/ramp/covid/simulation/parameters/PopulationParameters.java [hunk 6]
```diff
@@ -38,10 +38,10 @@ private PopulationParameters() {
 
     public boolean isValid() {
         ParameterInitialisedChecker checker = new ParameterInitialisedChecker();
-        boolean valid = true;
+        
         // We don't do this in a single statement to ensure that all the "uninitalised" parameter warnings are printed
         // in one go instead of being short circuited
```
[the remark this answers is not in the document]
Comment (BlairArchibald) [reply src/main/java/uk/co/ramp/covid/simulation/parameters/PopulationParameters.java:43]: Good spot - we probably need to decide if we want what it says in the comments, i.e. a big dump of all the errors, or if we would rather have the functionality that's implemented where you get a short list as an when they are found. As you say, easy enough to change in a future PR.

File: src/test/java/uk/co/ramp/covid/simulation/place/CareHomeTest.java (hunk 4 above)
Comment (BlairArchibald) [reply src/test/java/uk/co/ramp/covid/simulation/place/CareHomeTest.java:118]: I might explore this a bit more today first (to fix, or have a better idea what goes in the issue). I'm not sure it's necessarily an issue with the "standard" COVID progression; it's possible that we need to do slightly more when we "force" symptomatic cases (only in the tests).

If I track this down I'll make another PR so Paul can double check, else I'll make an issue.

File: src/main/java/uk/co/ramp/covid/simulation/place/CareHome.java (hunk 5 above)
Comment (BlairArchibald) [reply src/main/java/uk/co/ramp/covid/simulation/place/CareHome.java:17]: Looks like it arrived in one of the merge commits (4364c1). Not sure why github hasn't filtered it out (unless I had to change this here to fix the tests. Either way this is the correct variable to set, not the adjustment)

**Comment by johnnonweiler** [comment]
(I'm leaving this for @BlairArchibald to merge, because there may be some reason that he hasn't already merged it.)

**Pull request merged**